Lib/contextlib.py

```diff
@@ -441,7 +441,16 @@ def __exit__(self, exctype, excinst, exctb):
         # exactly reproduce the limitations of the CPython interpreter.
         #
         # See http://bugs.python.org/issue12029 for more details
-        return exctype is not None and issubclass(exctype, self._exceptions)
+        if exctype is None:
+            return
+        if issubclass(exctype, self._exceptions):
+            return True
+        if issubclass(exctype, ExceptionGroup):
+            match, rest = excinst.split(self._exceptions)
+            if rest is None:
+                return True
+            raise rest from excinst
```

Comment:
This isn't ideal as it makes the exception group's own traceback include the `def __exit__(...)` frames. For example:

```
  File "/Volumes/RAMDisk/cpython/Lib/test/test_contextlib.py", line 1220, in test_exception_groups
    with suppress(ValueError):
  File "/Volumes/RAMDisk/cpython/Lib/contextlib.py", line 454, in __exit__
    raise rest from excinst
  File "/Volumes/RAMDisk/cpython/Lib/test/test_contextlib.py", line 1221, in test_exception_groups
    raise eg_all()
```

in case of the newly added test.

This isn't a big problem because:
- the group's member exceptions contain pristine tracebacks, only the group itself contains extra frames; and
- the `raise` of the original group is still included in the traceback.

Ideally we wouldn't need this but the API of `__exit__` makes it impossible to replace the `ExceptionGroup` instance with another one, while `ExceptionGroup` itself makes its exceptions read-only.